tests/test_qhash.c

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+ * qLibc
+ *
+ * Copyright (c) 2010-2015 Seungyoung Kim.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *****************************************************************************/
+
+#include "qunit.h"
+#include "qlibc.h"
+
+QUNIT_START("Test qhash.c");
+
+FILE *f;
+size_t written;
+bool success;
+unsigned char digest[16];
+char *hash;
+const char *testfile = "hashme.txt";
+const char *testdata = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
```

Comment:
Also, have you checked if there are any common sample data pattern we can use for MD5 tests?

Reply:
I didn't see any data files in the tests directory to use. All the other tests generate their own data in memory. If you mean on the internet have I looked on the greater internet? I haven't looked. If we are good with a handful of binary data files, I would probably generate some 1024 bytes files from /dev/urandom or openssl -rand.  At this point, We probably aren't testing the MD5 algorithm itself, but just that the qlibc functions are doing what we expect?

Reply:
generating files from the random source sounds good to me.

Reply:
Also, The reason I chose an ascii string is because I can easily validate the unit tests with the command line utilities:
echo -n "bc" | md5sum
when testing qhashmd5_file(testfile, 1, 2, digest);
I'm open to recommendations for what kind of data to use for testing.

Reply:
I will add some small binary files to show working tests on non-ascii data.

Reply:
ok. I'm also fine with the way you implemented. I was thinking about future extension of testings but that's probably something that the next person to think about. Looks all good!